app/screens/send-bitcoin-screen/send-bitcoin-screen.tsx

```diff
@@ -236,10 +236,23 @@ export const SendBitcoinScreen: React.FC<SendBitcoinScreenProps> = ({
 
   useEffect(() => {
     reset()
-    const { valid, username } = validPayment(route.params?.payment, network, client)
+    const { valid, username, amount, memo, currency } = validPayment(
+      route.params?.payment,
+      network,
+      client,
+    )
     if (route.params?.username || username) {
       setInteractive(false)
       setDestination(route.params?.username || username)
+      if (currency === "USD") {
```

Comment:
Do you need this conversion for the new billpay? I don't think it will be called since you only pass payment and not username when navigating to route params.